src/main/java/com/microsoftopentechnologies/windowsazurestorage/helper/AzureUtils.java

```diff
@@ -169,6 +171,37 @@ public static SharedAccessBlobPolicy generateBlobPolicy() {
         return policy;
     }
 
+    public static String generateBlobSASURL(
+            StorageAccountInfo storageAccount,
+            String containerName,
+            String blobName,
+            EnumSet<SharedAccessBlobPermissions> permissions) throws Exception {
+
+        CloudStorageAccount cloudStorageAccount = getCloudStorageAccount(storageAccount);
+
+        // Create the blob client.
+        CloudBlobClient blobClient = cloudStorageAccount.createCloudBlobClient();
+        CloudBlobContainer container = blobClient.getContainerReference(containerName);
+
+        // At this point need to throw an error back since container itself did not exist.
+        if (!container.exists()) {
+            throw new Exception("WAStorageClient: generateBlobSASURL: Container " + containerName
+                    + " does not exist in storage account " + storageAccount.getStorageAccName());
+        }
+
+        CloudBlob blob = container.getBlockBlobReference(blobName);
+        String sas = blob.generateSharedAccessSignature(generateBlobPolicy(permissions), null);
+
+        return sas;
+    }
+
+    public static SharedAccessBlobPolicy generateBlobPolicy(EnumSet<SharedAccessBlobPermissions> permissions) {
```

Comment:
how about move all storage related operation to a new class called azureStorageClient.class? this can be  done in refactor later.

Reply:
Agree